ivadomed/loader/files_dataset.py

```diff
@@ -0,0 +1,239 @@
+        if config.metadata_choice == MetadataKW.MRI_PARAMS:
+            self.metadata: dict = {
+                "FlipAngle": [],
+                "RepetitionTime": [],
+                "EchoTime": [],
+                "Manufacturer": [],
+            }
+
+        if dict_files_pairing.get(DataloaderKW.MISSING_FILES_HANDLE) == FileMissingHandleKW.SKIP:
+            self.drop_missing = True
+
+        self.n_expected_input: int = int(dict_files_pairing.get(DataloaderKW.EXPECTED_INPUT, 0))
+        if self.n_expected_input <= 0:
+            raise ValueError("Number of Expected Input at File Dataset level must be > 0")
+
+        self.n_expected_gt: int = int(dict_files_pairing.get(DataloaderKW.EXPECTED_GT, 0))
+        if self.n_expected_gt <= 0:
+            raise ValueError("Number of Expected Ground Truth at File Dataset level must be > 0")
+
+        self.name: str = dict_files_pairing.get(DataloaderKW.SUBSET_LABEL, "DefaultFileDatasetLabel")
+
+        # We assume user explicitly provide the subject lists so WE do not do any additional filtering
+
+        # Currently does not support contrast balance (See BIDS Data 2D for reference implementation)
+        # Create a dictionary with the number of subjects for each contrast of contrast_balance
+
+        # NOT going to support bounding boxes
+
+        # NOT going to care about multi-contrast/channels as we assume user explicitly provide that.
+
+        # Get all derivatives filenames frm User JSON
+
+        #################
+        # Create filename_pairs
+        #################
+        self.path_data, self.filename_pairs = self.parse_spec_json_and_update_filename_pairs(dict_files_pairing)
+
+        if not self.filename_pairs:
+            raise Exception(
+                "No subjects were selected - check selection of parameters on config.json"
+            )
+
+        length = (
+            config.model_params[ModelParamsKW.LENGTH_2D]
+            if ModelParamsKW.LENGTH_2D in config.model_params
+            else []
+        )
+        stride = (
+            config.model_params[ModelParamsKW.STRIDE_2D]
+            if ModelParamsKW.STRIDE_2D in config.model_params
+            else []
+        )
+
+        # Call the parent class constructor for MRI2DSegmentationDataset
+        super().__init__(
+            self.filename_pairs,
+            length,
+            stride,
+            config.slice_axis,
+            config.nibabel_cache,
+            config.transform,
+            config.slice_filter_fn,
+            config.patch_filter_fn,
+            config.task,
+            config.roi_params,
+            config.soft_gt,
+            config.is_input_dropout,
+        )
+
+    def parse_spec_json_and_update_filename_pairs(self, loader_json: dict) -> Tuple[str, list]:
+        """Load the json file and return the dictionary"""
+        # Given a JSON file, try to load the file pairing from it.
+        assert loader_json.get(DataloaderKW.TYPE) == "FILES"
```

Comment:
To avoid typos & waste time on debugging:

`assert loader_json.get(DataloaderKW.TYPE).upper() == "FILES", f'Invalid {loader_json.get(DataloaderKW.TYPE)}!'`